packages/react-api/src/hooks/gpu-filter-utils.js

```diff
@@ -0,0 +1,88 @@
+import { _buildFeatureFilter, _FilterTypes } from '@carto/react-core/';
+
+// Don't change this value to maintain compatibility with builder
+export const MAX_GPU_FILTERS = 4;
+
+function getFiltersWithoutTimeType(filters) {
+  const filtersWithoutTimeType = {};
+  let timeFilter = {};
+
+  Object.entries(filters).forEach(([column, columnData]) => {
+    Object.entries(columnData).forEach(([type, typeData]) => {
+      if (type !== _FilterTypes.TIME) {
+        filtersWithoutTimeType[column] = { [type]: typeData };
+      } else {
+        timeFilter = {
+          column,
+          values: typeData.values,
+          type
+        };
+      }
+    });
+  });
+  return {
+    filtersWithoutTimeType,
+    timeFilter
+  };
+}
+
+function getFilterRange(timeFilter) {
+  const result = Array(MAX_GPU_FILTERS).fill([0, 0]);
+  // According to getFilterValue all filters are resolved as 0 or 1 in the first position of the array
+  // except the time filter value that is resolved with the real value of the feature in the second position of the array
+  result[0] = [1, 1];
+  if (Object.keys(timeFilter).length > 0) {
+    result[1] = timeFilter.values[0];
+  }
+  return result;
+}
+
+function getFilterValueUpdateTriggers(filtersWithoutTimeType, timeFilter) {
+  const result = { ...filtersWithoutTimeType };
+
+  // We don't want to change the layer UpdateTriggers evety time that the time filter changes
+  // because this filter is changed by the time series widget during its animation
+  // so we remove the the time filter value from the UpdateTriggers
+  if (Object.keys(timeFilter).length > 0) {
```

Comment:
And if we have more than 1 timeFilter? I think we should throw an error right?

Reply:
it's not possible according to our filter model. The `addFilter` action would overwrite the first time filter by the second